File: TME.Store/TME.Api.Domain/Models/ApiSearchResult.cs
using System;
using System.Collections.Generic;

namespace TME.Api.Domain.Models
{
    public class ApiSearchResult
    {
        public List<ApiSearchProduct> ProductList { get; set; }
        public int Amount { get; set; }
        public int PageNumber { get; set; }
        public Dictionary<string,int> CategoryList { get; set; }
    }
}

File: tme_store/lib/api/models/api_search_result.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:tme_store/api/models/api_poduct.dart';

part 'api_search_result.g.dart';

@JsonSerializable()
class ApiSearchResult {
  @JsonKey(name: 'Amount')
  final int totalProductsFound;

  @JsonKey(name: 'PageNumber')
  final int page;

  @JsonKey(name: 'ProductList')
  final List<ApiProduct> products;

  @JsonKey(name: 'CategoryList')
  final Map<String, int> totalProductsPerCategory;

  ApiSearchResult({
    this.totalProductsFound,
    this.page,
    this.products,
    this.totalProductsPerCategory
  });

  static ApiSearchResult fromJson(Map<String, dynamic> json) =>
      _$ApiSearchResultFromJson(json);

  Map<String, dynamic> toJson() => _$ApiSearchResultToJson(this);
}